inmem/kv.go

```diff
@@ -232,3 +239,92 @@ func (b *Bucket) getAll(o *kv.CursorHints) ([]kv.Pair, error) {
 
 	return pairs, nil
 }
+
+// ForwardCursor returns a directional cursor which starts at the provided seeked key
+func (b *Bucket) ForwardCursor(seek []byte, opts ...kv.CursorOption) (kv.ForwardCursor, error) {
+	pairs := make(chan []kv.Pair)
+	go func() {
+		defer close(pairs)
+
+		var (
+			config  = kv.NewCursorConfig(opts...)
+			fn      = config.Hints.PredicateFn
+			iterate = func(it btree.ItemIterator) {
+				b.btree.AscendGreaterOrEqual(&item{key: seek}, it)
+			}
+		)
+
+		if config.Direction == kv.CursorDescending {
+			iterate = func(it btree.ItemIterator) {
+				b.btree.DescendLessOrEqual(&item{key: seek}, it)
+			}
+		}
+
+		var batch []kv.Pair
+
+		iterate(func(i btree.Item) bool {
+			j, ok := i.(*item)
+			if !ok {
+				batch = append(batch, kv.Pair{Err: fmt.Errorf("error item is type %T not *item", i)})
+
+				return false
+			}
+
+			if fn == nil || fn(j.key, j.value) {
+				batch = append(batch, kv.Pair{Key: j.key, Value: j.value})
+			}
+
+			if len(batch) < cursorBatchSize {
+				return true
+			}
+
+			pairs <- batch
```

Comment:
This goroutine could leak (block forever) if the consumer of this cursor does not call `Next` to read all the data, which happens often when it reaches the last key it is interested in.

One approach that we use for the tsdb Cursor interface is to add a `Close()` API that is responsible for cleaning up any resources.

Reply:
good catch!